website/source/guides/security/vault-pki-integration.html.md

```diff
@@ -0,0 +1,550 @@
+next steps to generate our TLS certs, the recommended security approach is to
+create a new token based on a specific policy with limited privileges.
+
+Create a policy file named `tls-policy.hcl` and provide it the following
+contents:
+
+```
+path "pki_int/issue/nomad-cluster" {
+  capabilities = ["update"]
+}
+```
+Note that we have are specifying the `update` [capability][capability] on the
+path `pki_int/issue/nomad-cluster`. All other privileges will be denied. You can
+read more about Vault policies [here][policies].
+
+Write the policy we just created into Vault:
+
+```
+$ vault policy write tls-policy tls-policy.hcl
+Success! Uploaded policy: tls-policy
+```
+
+### Step 8: Generate a Token based on `tls-policy`
+
+Create a token based on `tls-policy` with the following command:
+
+```
+$ vault token create -policy="tls-policy" -ttl=24h
+```
+
+If the command is successful, you will see output similar to the following:
+
+```shell
+Key                  Value
+---                  -----
+token                s.xafiYzh7MCMotHLu2d35hepR
+token_accessor       9vj7q5nnF53JAcTyxvccpAZK
+token_duration       24h
+token_renewable      true
+token_policies       ["default" "tls-policy"]
+identity_policies    []
+policies             ["default" "tls-policy"]
+```
+
+Make a note of this token as you will need it in the upcoming steps.
+
+### Step 9: Configure Consul Template on All Nodes
+
+If you are using the AWS environment provided in this guide, you already have
+[Consul Template][consul-template-github] installed on all nodes. If you are
+using your own environment, please make sure Consul Template is installed. You
+can download it [here][ct-download].
+
+Provide the token you created in [Step
+8](#step-8-generate-a-token-based-on-tls-policy) to the Consul Template
+configuration file located at `/etc/consul-template.d/consul-template.hcl`. You
+will also need to specify the [template stanza][ct-template-stanza] so you can
+render each of the following on your nodes at the specified location from the
+provided templates (the actual templates will be provided in the next step):
+
+* Node certificate
+* Node private key
+* CA public certificate
+
+We will also specify the template stanza to create certs and keys for the Nomad
+CLI (which defaults to HTTP but will need to use HTTPS once once TLS is enabled
+in our cluster).
+
+Your `consul-template.hcl` configuration file should look similar to the
+following (you will need to provide this to each node in the cluster):
+
+```
+vault {
+  address      = "http://active.vault.service.consul:8200"
+  token        = "s.xafiYzh7MCMotHLu2d35hepR"
+  grace        = "1s"
+  unwrap_token = false
+  renew_token  = true
+}
+
+syslog {
+  enabled  = true
+  facility = "LOCAL5"
+}
+
+template {
+  source      = "/opt/nomad/templates/agent.crt.tpl"
+  destination = "/opt/nomad/certs/agent.crt"
+  command     = "pkill -SIGHUP nomad"
```

Comment:
Can we use `systemctl reload nomad` instead? I *think* pkill will send SIGHUP to the subprocesses nomad spawns that also start with "nomad" like executor, logmon, and docker_logger.

I think all of those processes ignore SIGHUP, but it seems safer to use systemd.

Reply:
Implemented by kill in the example systemd unit 
https://github.com/hashicorp/nomad/blob/4ca6cda6c1077474a3cde93c9b7ed5ac374eec35/dist/systemd/nomad.service#L14

Reply:
The systemd command only sends the signal to the main Nomad process.

AFAIK pkill will happily send the signal to any process starting with "nomad" which includes a lot of helper subprocesses.

Reply:
Ah. good point!  I wasn't thinking about pkill being "helpful".  Apparently, that's selectable behavior with `-x`, but I had `pidof` behavior in my head which is less generous by default.

One side question though: systemctl requires that you are using systemd as your init.  kill is basically a promise.  I'm fine either way, but incorporating systemctl specifically in the guide does break it for _some_ subset of people.